sessionctx/variable/session.go

```diff
@@ -1281,6 +1281,84 @@ type SessionVars struct {
 	// OptPrefixIndexSingleScan indicates whether to do some optimizations to avoid double scan for prefix index.
 	// When set to true, `col is (not) null`(`col` is index prefix column) is regarded as index filter rather than table filter.
 	OptPrefixIndexSingleScan bool
+
+	MaxReuseChunk  int
+	MaxReuseColumn int
+	// ChunkPool Several chunks and columns are cached
+	ChunkPool struct {
+		Lock  sync.Mutex
+		Alloc chunk.Allocator
+	}
+	EnableReuseCheck bool
+
+	// useChunkAlloc indicates whether statement use chunk alloc
+	useChunkAlloc    bool
+	preUseChunkAlloc bool
+}
+
+// GetNewChunk Attempt to request memory from the chunk pool
+// thread safety
+func (s *SessionVars) GetNewChunk(fields []*types.FieldType, capacity int) *chunk.Chunk {
+	//Chunk memory pool is not set
+	if s.ChunkPool.Alloc == nil {
+		return chunk.NewChunkWithCapacity(fields, capacity)
+	}
+	s.ChunkPool.Lock.Lock()
+	defer s.ChunkPool.Lock.Unlock()
+	if s.checkReuseAllocSize() && (!s.useChunkAlloc) {
+		s.useChunkAlloc = true
```

Comment:
Seems the `useChunkAlloc` would be reset for each statement, maybe it's more appropriate to put it into the `stmtContext`. and use it in the later statement processing.

Reply:
done